spot-client/src/common/app-state/remote-control-service/actions.js

```diff
@@ -209,6 +209,21 @@ export function startWirelessScreensharing() {
     });
 }
 
+/**
+ * Cleans ups any wireless screensharing connection being established or already
+ * established.
+ *
+ * @returns {Function}
+ */
+export function forceStopWirelessScreenshare() {
+    return dispatch => {
+        remoteControlService.destroyWirelessScreenshareConnections();
+
+        dispatch(setLocalWirelessScreensharing(false));
```

Comment:
does it duplicate the code from `stopScreenshare` ?

Reply:
stopScreenshare calls destroyWireless. stopScreenshare does the XMPP command messaging, whereas destroyWireless does not so its cleanup is all local. I don't know why I thought the difference was important, maybe it isn't.

Reply:
I mean those two lines 222 and 223

Reply:
Same answer. I'll have to look into it more.

Reply:
Answer is no they do different things but it can probably be refactored so there isn't redundancy.

Reply:
yes I know they do different things but the logic for resetting the state is duplicated

Reply:
It looks like local wireless screenshare state is in one reducer while spot-tv state is another reducer. setLocalWirelessScreensharing updates local state while setSpotTVState is called to fake an immediate response from Spot-TV about its state update. I'm going to remove the setSpotTVState call because it should not be needed for this case.